### ISSUE TITLE
Got many errors like "Conflicting types for `gpr_event_init`" when building RouteGuideClient in Xcode 7.1

### ISSUE THREAD

**Issue by kudocc** [grpc/grpc#4027]
I follow the guide [here](http://www.grpc.io/docs/tutorials/basic/objective-c.html), but get many compile errors. Please see the picture below.

![compileerror](https://cloud.githubusercontent.com/assets/4947330/10903193/07d68672-8242-11e5-9fbe-d47cddb2cf47.png)

**Comment by jcanizales** [comment]
Hi Kudo,

I'm sorry for this difficulty. I'm not able to reproduce your problem (I'm using XCode 7.1 too); building gives me just warnings that are being worked on:
<img width="261" alt="screen shot 2015-11-04 at 9 58 07 pm" src="https://github-cloud.s3.amazonaws.com/assets%2F5120183%2F10961081%2F90e06372-833f-11e5-8926-1a90bee45ef8.png">

To help me debug the issue, could you tell me what's the diagnostic information given by XCode when you open any of the errors shown (by clicking the triangle)?

Also, to double check, after you do `pod install`, the workspace created should have these folders and products:
<img width="221" alt="screen shot 2015-11-04 at 10 05 08 pm" src="https://github-cloud.s3.amazonaws.com/assets%2F5120183%2F10961149%2F6d207a20-8340-11e5-96a7-7405264b1aad.png">
Is that what you have?

Cheers,
Jorge

**Comment by kudocc** [comment]
Hi, Jorge

Thanks for your comment. In fact, I got other compile error at the beginning, you can see it in this picture.
![git](https://cloud.githubusercontent.com/assets/4947330/10961646/f2d6067c-83cb-11e5-8007-96164e149bb7.png)

It seems that the compiler doesn't find the definition of `time_t`, so I include a header file by adding `#include <sys/time.h>` below the line `#include <time.h>` in "/include/grpc/support/time.h". I build it again and I get the errors like I described in the issue. Please see my pictures.

![git2](https://cloud.githubusercontent.com/assets/4947330/10961910/621ce9ea-83ce-11e5-8c03-e7b81d7a53c7.png)

![git3](https://cloud.githubusercontent.com/assets/4947330/10961913/6d980890-83ce-11e5-8b3d-7303cc2067ac.png)

This is the architecture of my workspace.

![git4](https://cloud.githubusercontent.com/assets/4947330/10961936/a5134d84-83ce-11e5-9fb7-6d4ee6fcf2cf.png)

**Comment by kudocc** [comment]
Hi, Jorge

After I remote the `grpc` directory and redo the whole process, the issue is gone. There is only a link error which is about Bitcode. After I disable the Bitcode in `RouteGuideClient` Build settings, build success.

> ld: '/Users/kudocc/Desktop/git/grpc/examples/objective-c/route_guide/Pods/OpenSSL/lib/libcrypto.a(bio_lib.o)' does not contain bitcode. You must rebuild it with bitcode enabled (Xcode setting ENABLE_BITCODE), obtain an updated library from the vendor, or disable bitcode for this target. for architecture arm64

I think it must be something I have done messed it up.

Thanks for your time. Many thanks :+1:

**Comment by jcanizales** [comment]
The problem XCode 7 has with `time.h` should have gone away on the master branch after PR https://github.com/grpc/grpc/pull/3990 (and on the release 0.11 branch after PR https://github.com/grpc/grpc/pull/4011). Can you verify that you're checked out past that commit?

**Comment by kudocc** [comment]
My Podfile is

>  pod 'Protobuf', '~> 3.0.0-alpha-4'
>  pod 'gRPC', '~> 0.11.1'

It shows :
>Installing OpenSSL (1.0.204.1)
>Installing Protobuf (3.0.0-alpha-4.1)
>Installing gRPC (0.11.1)

I created another Project and pod install again. Now the errors have gone, thanks Jorge : )

**Comment by jcanizales** [comment]
No problem! Please let us know how the library works for you!

**Comment by kudocc** [comment]
Yes, I will do after using it for some time.

**Comment by jacobkneal** [comment]
I'm having the opposite issue--works fine in simulator but errors out on real device: 
<img width="297" alt="screen shot 2016-04-07 at 5 50 13 pm" src="https://cloud.githubusercontent.com/assets/869528/14371121/40201030-fce9-11e5-9944-2ba7789141f7.png">

**Comment by jacobkneal** [comment]
I'm installing from source: 
`
pod 'Google/SignIn'
pod 'SWTableViewCell', '~> 0.3.7'

source 'https://github.com/CocoaPods/Specs.git'
platform :ios, '8.0'

pod 'Protobuf', :path => "../grpc/third_party/protobuf"
pod 'BoringSSL', :podspec => "../grpc/src/objective-c"
pod 'gRPC', :path => "../grpc"

# Depend on the generated AuthMessageService library.
pod 'AuthMessageService', :path => '.'

`

**Comment by jorgeolivero** [comment]
I'm also experiencing this issue when running on a device, but _not_ when running through simulator. 

Picture of build errors:

<img width="1552" alt="screen shot 2016-04-23 at 12 32 53 pm" src="https://cloud.githubusercontent.com/assets/3119712/14762722/d1793f94-094f-11e6-99e0-52b4a2f9fbdf.png">

And the podspec which generates the client:

```
Pod::Spec.new do |s|
  s.name     = 'octopus'
  s.version  = '0.0.1'
  s.license  = '...'

  s.ios.deployment_target = '7.1'
  s.osx.deployment_target = '10.9'

  # Run protoc with the Objective-C and gRPC plugins to generate protocol messages and gRPC clients.
  # You can run this command manually if you later change your protos and need to regenerate.
  s.prepare_command = "protoc --objc_out=. --objcgrpc_out=. --proto_path=../../../proto ../../../proto/*.proto"

  # The --objc_out plugin generates a pair of .pbobjc.h/.pbobjc.m files for each .proto file.
  s.subspec "Messages" do |ms|
    ms.source_files = "*.pbobjc.{h,m}"
    ms.header_mappings_dir = "."
    ms.requires_arc = false
    ms.dependency "Protobuf", "~> 3.0.0-alpha-4"
  end

  # The --objcgrpc_out plugin generates a pair of .pbrpc.h/.pbrpc.m files for each .proto file with
  # a service defined.
  s.subspec "Services" do |ss|
    ss.source_files = "*.pbrpc.{h,m}"
    ss.header_mappings_dir = "."
    ss.requires_arc = true
    ss.dependency "gRPC", "~> 0.12"
    ss.dependency "#{s.name}/Messages"
  end
end
```

**Comment by jcanizales** [comment]
Is there a way for either of you guys to share a project with us, to reproduce this issue? It's so weird that I'm afraid there's not much I can do if I can't actually look at it.

**Comment by jorgeolivero** [comment]
@jcanizales yeah, sure. I'll send you a zip, which email do you prefer?

**Comment by jcanizales** [comment]
My username, at google dot com, thanks!

**Comment by jorgeolivero** [comment]
@jcanizales sent, thanks for the help

**Comment by jorgeolivero** [comment]
<img width="1552" alt="screen shot 2016-04-27 at 8 42 00 am" src="https://cloud.githubusercontent.com/assets/3119712/14852385/f86d3f68-0c53-11e6-8bac-0dbde8ae314a.png">

Here's some more context on the build errors,  It seems that those types have gotten defined twice

**Comment by jorgeolivero** [comment]
@jacobkneal Were you able to address the issues?

**Comment by jcanizales** [comment]
Hey Jorge - sorry, I haven't had time to look into your project yet. We're working full time on fixing the problems with Swift+Cocoapods compatibility.

**Comment by jorgeolivero** [comment]
@jcanizales Good to hear, I was able to unblock myself with some input from a user in another issue

**Comment by danmatthews** [comment]
@jorgeolivero sorry to message you almost 2 years on, but how did you solve it? I'm having a similar issue over here https://github.com/invertase/react-native-firebase/issues/758#issuecomment-362078291.

**Comment by jcanizales** [comment]
@muxi

**Comment by arnolddrita** [comment]
I am also having this problem and I would really need a solution for this. 
<img width="1155" alt="screen shot 2018-02-20 at 4 13 15 pm" src="https://user-images.githubusercontent.com/18261031/36431887-11ce1b38-1659-11e8-8858-5fe69b44cf47.png">

**Comment by danmatthews** [comment]
@arnolddrita can you post your `Podfile` please?

**Comment by jdkanani** [comment]
Same error as @arnolddrita when I add `react-native-camera` and do manual ios settings.

**Comment by muxi** [comment]
Just notice this thread and sorry for having missed.

@jdkanani Could you share more information about your project and how we can reproduce?

@danmatthews - Your issue looked like a known issue in Cocoapods 1.3 and 1.4. Version 1.5.0 should have resolved this problem. Hopefully you have resolved this problem already.

**Comment by arnolddrita** [comment]
I don't really know what did it, but I just removed the library and re-added it from the start, and it worked the second time.

**Comment by jackcbrown89** [comment]
Any updates on this? I get the exact same errors as @arnolddrita when I run in debug scheme. Release scheme builds fine.

**Comment by muxi** [comment]
Could you provide some info on how to reproduce? I looked into this issue but wasn't able to see it. Our tests seem to be running well too.

**Comment by jackcbrown89** [comment]
Thanks for the speedy reply! After trying to reproduce the issue, I've found that it's probably just something with my configuration for my pods that are used in the debug scheme because I couldn't create the error in a new project. If I switch the debug scheme to use the release pods, the simulator runs fine, but I can't push it to a real device without the error occuring. Any ideas?

I'll delete the build folder and see if that helps. Will write back what happens.
